src/ServiceResolver/Resolvers/MemoryCache/CachedDefinition.php

```diff
@@ -19,9 +19,10 @@ public function __construct(
 
     public function evaluate(PsrContainer $psrContainer): object
     {
-        return match ($this->cache) {
-            null => $this->cache = $this->decorated->evaluate($psrContainer),
-            default => $this->cache,
-        };
+        if (null === $this->cache) {
+            $this->cache = $this->decorated->evaluate($psrContainer);
+        }
```

Comment:
Not actually suggesting to change this, but would this `if` block be equivalent to the code below?
```php
$this->cache ??= $this->decorated->evaluate($psrContainer);
```

Reply:
Resolved here: baef19c